server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -459,24 +597,314 @@ else if (ownerDto.getKey() != null) {
         }
     }
 
-    private Pool findPool(String poolId) {
-        Pool pool = poolManager.find(poolId);
+    /**
+     * Populates the specified entity with data from the provided DTO.
+     *
+     * @param entity
+     *  The entity instance to populate
+     *
+     * @param dto
+     *  The DTO containing the data with which to populate the entity
+     *
+     * @throws IllegalArgumentException
+     *  if either entity or dto are null
+     */
+    @SuppressWarnings("checkstyle:methodlength")
+    protected void populateEntity(Pool entity, PoolDTO dto) {
```

Comment:
We need to step through this a bit more carefully and ignore fields we don't allow the client to set.